syft/messaging/plan.py

```diff
@@ -562,15 +576,30 @@ def __call__(self, *args, **kwargs):
             raise ValueError("Kwargs are not supported for plan.")
 
         if self.find_location(args) == self.owner:
-            if self.forward is not None:
+            if any(
+                [hasattr(arg, "child") and isinstance(arg.child, PromiseTensor) for arg in args]
+            ):
+                return self.setup_plan_with_promises(*args)
+            elif self.forward is not None:
                 if self.include_state:
                     return self.forward(*args, self.state)
                 else:
                     return self.forward(*args)
             else:
                 return self._execute_readable_plan(*args)
         else:
-            if not self.is_built:
+            if any(
+                [
+                    hasattr(arg, "child")
+                    and isinstance(
+                        arg.child.location._objects[arg.child.id_at_location].child, PromiseTensor
```

Comment:
That's not allowed ;)
But as discussed it's out of scope for this PR, maybe juste remove it